### DISCUSSION TITLE
uBO Minus/Lite discussion

### DISCUSSION THREAD

**Discussion by Yuki2718** [uBlockOrigin/uBlock-issues#2261] [General]
https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-1242401687
> if you have a better qualifier than Minus, I welcome suggestions.

IMHO it's best to assign a new name and my first suggestion is uPrivacy. Obviously I have no right to define what uBO is, but as an old user since the very first version of HTTPSB, I don't see how uBO without dynamic filtering can be uBO. It's one of the unique characteristics and identity of uBO among many such as superior performance. Well, if I understand correctly neither performance will be a unique characterisitics in MV3 as now filtering is done by browser. gorhill's [statement](https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-456179825)
> I won't work to make uBO less than what it is now.

can be pro-argument to assign a new name. Whatever the intention was, in my country it turned out people's perception is: "Oh, relieved. So I can use uBO in MV3 world, though some function may be restricted." I don't think this is caused by the name, yet still it's confusing. And as gorhill had to clarify, others take "Minus" as a sign of negative feelings. My rationale for the name uPrivacy is:

- uBOM still keeps one characteristics of uBO: blocking tracker by default (other major blocker not).
- it doesn't require read-data permission which can potentially undermine user privacy.

Other candidates I think of are uFilter, uDefender, uProtector, etc. but I feel uPrivacy best fit for the reason above.

[Update] uBO's [manifest](https://github.com/gorhill/uBlock/blob/master/MANIFESTO.md):
> uBlock's sole purpose is to give users the means to enforce their own choices.

I don't see how this is possible without dynamic filtering. Now it's not possible to block XSS through iframe, to block watering hole attack with 3p script, and even *100%* blocking Facebook anywhere except sites user chose is not trivial even if uBOL supported My filters and custom subscription. TBH uBOL looks to be just "ad (and tracking) blocker" which uBO denies to be.

**Comment by mapx-** [comment]
https://github.com/gorhill/uBlock/commit/93e5133783301c0329b3ce8f9e6079badd06d62d
![image](https://user-images.githubusercontent.com/10303732/189630355-0fdf5b21-c499-4c34-b926-9e6a91784649.png)

**Reply by Yuki2718** [reply]
My point still holds.

**Reply by MasterKia** [reply]
I wonder, why was **Minus** perceived as "controversial"?

Users should know that Chromium (_Google_) crippled content blockers and **Minus** aptly implies that, "uPrivacy" doesn't.

As gorhill [has said before](https://github.com/uBlockOrigin/uBlock-issues/issues/1553#issuecomment-1032762806), Google isn't serving the user agent anymore.

**Reply by peace2000** [reply]
I think uBO should be loud and clear on Chrome web store that Chrome version is lesser than true uBO (and reasons why). It should even mention other browsers in description in which uBO works for it's best potential and suggest users to migrate to them. Some people won't care and will still install lesser uBO but some people could change their browser.

Just removing uBO from Chrome web store or changing it's name to something totally different would IMO be kind of hiding. Then people would not become aware of Google's mischiefs. I think being upfront and loud would be the best option. uBO should not back down from Chrome web store.

uBock Origin Lite is probably a good name.

**Reply by krystian3w** [reply]
If possible paste links without ban addon, link can protect long instruction to extend "quota" in textarea or censorship by Google team "to check description and code".

**Reply by ghost** [reply]
Not just a good name, but also makes it a better abbreviation "**UBOL**" 🤌

**Reply by garry-ut99** [reply]
> Yuki2718: Other candidates I think of are uFilter, uDefender, uProtector, etc. but I feel uPrivacy best fit for the reason above.

I know I'm late to the discussion, anyway just have wanted to put my 5 cents:
- uProtector too similiar to: uBlockProtector: (NSFW) `https://github.com/jspenguin2017/uBlockProtector` and
- uDefender too similiar to: NanoDefender: (NSFW) `https://jspenguin2017.github.io/uBlockProtector/`

**Comment by Yuki2718** [comment]
@MasterKia @peace2000  The extension name should not be used politically to accuse or critisize anyone (or Google). @gorhill 's *public* attitude toward MV3 has always been neutral and he's always seeking for what can be done on MV3. Whether Google's move is good or not is what each user should decide. If one is happy with uBOL's capability, it's pure improvement over uBO. What I said is simple, uBO without dynamic filtering is, *IMO*, **NOT** uBO. Sure, only gorhill can define what uBO is, but its origin is HTTPSB which gives noob user like me at the time full control over what requests are allowed. Static filtering was a later addition as PoC and even then gorhill used to recommend using ABP alongside with it. It evolved over time to the point I can't say dynamic filtering is the sole identity of uBO, but still lacking it means loosing roughly 50% of what it is.

**Reply by peace2000** [reply]
I think that using the name "Lite" isn't an accuse or criticism, rather a statement. Just like high-end mobile phones have their cheaper "Lite" versions. Minus could be seen as a negative name but I think Lite isn't that negative.

I'm glad that gorhill tries to find solutions for MV3 Chrome, I was worried that MV3 Chrome isn't going to see uBO at all. But still, people should not think that MV3 uBO is equal to MV2 uBO. Removing all uBlock Origin named addons from Chrome web store would be like leaving people in the dark (where is uBO? Etc....).

What it comes to dynamic filtering, I think that masses (regular people who just want an adblocker to block their ads via install-and-forget approach) won't really need it. Advanced users who use dynamic filtering are knowledgeable enough to switch to other browsers. Personally, I don't even use dynamic filtering that often, very rarely actually. For me, static filtering is more important aspect of uBO.

**Reply by MasterKia** [reply]
> gorhill 's public attitude has always been neutral

https://github.com/uBlockOrigin/uBlock-issues/issues/1553#issuecomment-1032762806:
> For years I've pointed out that _Chrome is an adtech agent_, **_not a user agent_**, however hard they try to spin it as the latter. This ever present motivation to have adtech API embedded in it just further illustrate that point.

Doesn't look "neutral" to me.

**Reply by Yuki2718** [reply]
That's about FLoC. By emphasizing *public* I mean whatever his personal thought on the matter is, his actual activities on the matter is always fact-based than belief or personal preference as seen in the mega thread. The oppsite is possible: "I believe MV3 is bad/I dislike Google, so I don't do anything for them" (or even worse only complain and critisize).

**Reply by MasterKia** [reply]
>  If one is happy with uBO Lite's capability, it's pure improvement over uBO.

Who's gonna be "happy" with:
- Big ads placeholders <sub>_No cosmetic filtering (##)_</sub>
- Anti-Blocker warning <sub>No scriptlet injection (##+js)</sub>
- Third-party exposure due to exception instead of redirect <sub>No redirect= filters</sub>
- URL Tracking <sub>No removeparam= filters</sub>

?

**Reply by Yuki2718** [reply]
You should not take your view point as an absolute one. There are so many people happy only with DNS filtering. (BTW I personally almost never care placeholder actually.)

**Reply by MasterKia** [reply]
> I personally almost never care placeholder

"Ads" placeholder is the tip of the iceberg, placeholder can also be annoyance: force login, soft anti-blocker, newsletter sign up, social buttons.

(I don't care about the last two).

**Reply by Yuki2718** [reply]
IIRC Vivaldi blocker also doesn't (possibly "didn't") support cosmetic filtering and many people are satisified with it. BTW how about waiting for gorhill's reply before it becomes too long to read?

**Reply by MasterKia** [reply]
> You should not take your view point as an absolute one

I didn't; I just looked at all the detection/nuisance reports in the uAssets.
And for my "personal" view, I have had reports in my list about anti-blocker and site timer (both requiring scriptlet fixes).

**Reply by krystian3w** [reply]
Vivaldi support vanilla hide elements on pages by EasyList cosmetic syntax, also I saw added subscribe custom lists (maybe on Android too), so needed test procedural filtering (Brave don't have in Shields), implement `:style`, scriptlets, redirect and removeparam.

**Reply by Yuki2718** [reply]
Yep, confirmed supporing simple cosmetic filters. No support for procedural, style, scriptlet, or redirect tho.

**Reply by gorhill** [reply]
> Who's gonna be "happy" with:

The extension is to be permission-less at install. I do plan into looking at letting users explicitly grant uBO superior powers on whatever sites they choose to, either one, a handful, or everywhere. These are are called [optional_host_permissions](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#host-permissions). I started to work on that MV3 stuff less than 2 weeks ago, so people will have to accept it it won't have everything which can be had right now, there will be a progression.

**Reply by uBlock-user** [reply]
> These are are called [optional_host_permissions](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#host-permissions)

Sounds like in Android where a app is by default always bare-bones and post-install you start enabling permissions to make certain features work.

**Reply by krystian3w** [reply]
More iOS, because many Android apps can crash without allow insane premissions.

**Reply by ghost** [reply]
Regarding permissions, it's clear that MV3 tends to be more secure interms of extension control. Google themselves know how convoluted the unknown extensions lurking around the web store, ready to hijack user's privacy & security.
They had to implement these restrictions no matter what the circumstances. It kinda let's say not good for content blockers for leaving us a restricted approach, but Google knows how smart they are by allowing more adverts & content that benefit web journals, articles & hosts for their overall increasing revenue.

**Comment by lewisje** [comment]
Why does the manifest say that it requires at least version 105? I thought that MV3 was available in earlier versions of Chromium, although some of the new MV3 APIs that this extension relies on might not be.

My problem is that I couldn't install it in Brave because I use a flag that puts a "99." in front of the version in my User-Agent string, just in case some old websites think I'm using a version 10 or 1 browser; it's not a big deal, because I could just turn that flag off (I also noticed that Bloomberg BusinessWeek gives me an old-browser warning precisely because the first number in the version is reported as 99).

**Reply by krystian3w** [reply]
So use more soft UA overrider (to don't touch Chrome Web Store), also check how override client hints if Google and Brave don't disabled this in flags:

https://developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API

Because with the client hints, they read whether Brave is using Chromium 105+.

It probably won't be long before this flag needs to be reprogrammed as sites will start to recognise Chromium 99 as realistically too old in 2023.

---

Also unpacked zip or crx (compressed) should not check version by user agent.

**Reply by gorhill** [reply]
> Why does the manifest say that it requires at least version 105?

Because I don't want issues with the `:has()` operator found in filter lists -- there is no way with a declarative approach to deal specially with `:has()`. As of version 105, `:has()` is natively supported by Chromium.

**Reply by lewisje** [reply]
> So use more soft UA overrider (to don't touch Chrome Web Store), also check how override client hints if Google and Brave don't disabled this in flags:
> 
> [developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API](https://developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API)
> 
> Because with the client hints, they read whether Brave is using Chromium 105+.
> 
> It probably won't be long before this flag needs to be reprogrammed as sites will start to recognise Chromium 99 as realistically too old in 2023.
> 
> Also unpacked zip or crx (compressed) should not check version by user agent.

The Chrome Web Store apparently *does* check the `User-Agent`, and that's where I tried to install the extension from.

**Reply by krystian3w** [reply]
So move this to UA overrider add-on and monthly update brave numbers at end of user-agent.

**Reply by lewisje** [reply]
Oh, an easier workaround is to stop using that flag, because Brave does not prepend `99.` to the version string by default; I just thought it was a good idea to point out that the flag actually leads to a problem when trying to install the extension from the official source.

**Reply by krystian3w** [reply]
I think add-ons in brave also can't touch web store, but mv3 firewall block something by network rules - maybe Google fail.

**Reply by lewisje** [reply]
I just bothered to actually try disabling the #force-major-version-to-minor flag to see what would happen, and indeed, uBO Lite is marked as "Not Compatible" when the flag is enabled, and it is able to be installed when the flag is disabled.

This seems like a WONTFIX because there is a legitimate reason for setting 105 as the minimum Chromium version, and the Chrome Web Store is behaving exactly as expected upon seeing Brave sending a version string starting with `99.`; also, this has nothing to do with the fact that extensions cannot run on the Chrome Web Store itself, because this behavior was modified by a flag rather than by an extension.

**Comment by dportvine** [comment]
@gorhill 
With uBlock Origin Lite i see ads on this site `https://www.ukr.net/` (need Ukrainian IP)
But with uBO 1.44.4 (default + RU AdList) i don't see any ads.  It can be fixed?
With AdGuard MV3 i don't see any ads too.
[Gif](https://imgur.com/tXMMEch)

**Reply by mapx-** [reply]
see
https://github.com/gorhill/uBlock/blob/master/dist/mv3/log.txt

search for `ukr.net`

You'll get 3 invalid regex and 3 invalid redirecting filters

AG is using other filters for that site => it's why is still working

maybe @dimisa-RUAdList could replace those filters

**Reply by dportvine** [reply]
These are not three regular expressions, but the same one mentioned three times. In addition, it does not work on `ukr.net`, and this can be seen from domain=. And the `informers.ukr.net` domain is registered as an exception, i.e. it should not be blocked on the target domains, which are specified just in domain=.

The only reason why ukr.net uses redirect rules is because uBO doesn't know how to handle :remove() correctly in the background tab, so when you're on `https://www.ukr.net/news/main.html` and opening the main page in a new tab - we get unblocked ads. These invalid redirect rules do not directly affect the operation of :remove().

**uBO 1.44.4 only default**, add in my filters: 
```adb
ukr.net##body > iframe[name]:not([id]):not([class]):not([src]):remove()
@@||s.znctrack.net/z|$third-party,xmlhttprequest
```
**Important you need Ukrainian IP**

[Gif](https://imgur.com/UOAND52)

<details>
<summary>Screen</summary>

![2022-11-10_203430](https://user-images.githubusercontent.com/111344219/201178687-fc366e95-359a-4a36-ac99-64f8711aa6c9.png)
</details>

@gorhill

**Reply by mapx-** [reply]
in uBO - using ukr ip - I get no side ad.
Why did you add those 2 filters ?

**Reply by dportvine** [reply]
To hide zmctrack.net or znctrack.net ads
RU AdList - disabled?

**Reply by mapx-** [reply]
No, RU AdList enabled

**Reply by dportvine** [reply]
uBO 1.44.4 only default

**Reply by mapx-** [reply]
wait, I enabled RU AdList in uBOL and I get no side ad now

**Reply by dportvine** [reply]
Refresh the page several times. I'm not lying. (Ukrainian IP)
[Gif](https://imgur.com/KWYPRim)

**Reply by dportvine** [reply]
For me, these rules don't work in uBOL, but they do work in uBO. They hide ads. That's what I'm trying to convey.
The problem is in the uBOL.

**Reply by gorhill** [reply]
> `ukr.net##body > iframe[name]:not([id]):not([class]):not([src]):remove()`

I don't have a Ukraine address IP but I can see what I guess are ads when using a Latvia IP address.

Procedural filters are applied at [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) time, and probably it does not fire when the page is not visible, which is good since we do not want procedural filters to consume CPU cycle if the result is not going to be visible anyway. By the time the procedural filter is effectively applied, the page layout has changed and the `remove()` filter no longer matches.

In any case, vaguely reporting an issue as a review in the CWS which I have little chance of ever reading is not helpful. What helps is to report such issues here so they can be looked into.

**Reply by mapx-** [reply]
Tested again.

uBOL: if I disable the filtering I get these ads

![image](https://user-images.githubusercontent.com/10303732/201185371-ead0eed3-37cc-4d35-9949-45415fe1a6d2.png)

If I enable uBOL I get no such ads even refreshing repeatedly the page.

Using uBlock Origin Lite 0.1.22.11056

Maybe you are seeing other ads and my vpn ukr ip is detected

**Reply by dportvine** [reply]
Okay, what more information do you need? Can you fix it?

**Reply by gorhill** [reply]
I can reproduce the same issue in both uBO and uBOL. The ads do not show up in uBO/uBOL and loading the page in full view, and the ads show up in both uBO/uBOL when opning the page in the background.

**Reply by gorhill** [reply]
@mapx-, I get the ads in both uBO/uBOL when I middle click the site's logo then bring forth the new tab being opened (maybe wait one second before activating the tab).

**Reply by mapx-** [reply]
hmmm, right, middle clicking the logo brings back the ads

**Reply by dportvine** [reply]
@mapx- 
if it's not like this, your IP address is invalid
<details>
<summary>Screen</summary>

![2022-11-10_211704](https://user-images.githubusercontent.com/111344219/201186575-48432a11-4d93-48d7-bb9e-253358e287e0.png)
</details>

**Reply by mapx-** [reply]
wait, it's something weird:

only using uBO => no ad even middle clicking
using only uBOL or uBOL + uBO => I'll get ads using middle-clicking

**Reply by dportvine** [reply]
I see ads without middle click! In uBOL.

**Reply by mapx-** [reply]
@dportvine it's about ads like these:
![image](https://user-images.githubusercontent.com/10303732/201188027-c7810a77-d066-4323-a0a6-b9be50125773.png)

**Reply by gorhill** [reply]
uBOL hides the ads properly on my side, unless the tab is opened in the background, in which case the `:remove()` filter no longer matches the ads on the page because by then they have been moved into a closed shadow root.

**Reply by dportvine** [reply]
With RU AdList in uBO:

>The only reason why ukr.net uses redirect rules is because uBO doesn't know how to handle :remove() correctly in the background tab, so when you're on `https://www.ukr.net/news/main.html` and opening the main page in a new tab - we get unblocked ads.

**Reply by dportvine** [reply]
> uBOL hides the ads properly on my side, unless the tab is opened in the background, in which case the `:remove()` filter no longer matches the ads on the page because by then they have been moved into a closed shadow root.

But in uBO i don't have this problem.

**Reply by gorhill** [reply]
I can reproduce in uBO. What filter lists are you using in uBO?

**Reply by dportvine** [reply]
RU AdList

**Reply by gorhill** [reply]
Anyway, I can reproduce at will and I am pretty sure I understand the issue, which is not that `:remove()` is not supported, but it's uBO/uBOL uses `requestAnimationFrame()` to launch cycles of procedural filter application, and `requestAnimationFrame()` probably does not fire when the whole page is not visible. That page at `ukr.net` does something after being fully loaded, it moves the ads into a shadow root, which means the `remove()` filter no longer matches by the time a procedural filter application is launched.

**Reply by dportvine** [reply]
Can this be fixed for uBOL?

**Reply by gorhill** [reply]
Maybe I made a mistake when testing, I can't reproduce anymore with uBO, but I can reproduce with uBOL or with uBO/uBOL. Maybe uBOL was still enabled when I thought it was disabled. So now I am thinking the issue is with the missing `redirect-rule` filters which are not supported in uBOL. I need to investigate more...

**Reply by gorhill** [reply]
Ok I found the problem, it's much simpler than I thought. This will be fixed in next release.

**Comment by dimisa-RUAdList** [comment]
@gorhill 
>Procedural filters are applied at [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) time, and probably it does not fire when the page is not visible, which is good since we do not want procedural filters to consume CPU cycle if the result is not going to be visible anyway.

This reasoning is understandable. However, the page opened in the background tab will still be visited by the user in the end. And there will be unblocked ads on this page.

To avoid this, for those sites where the `remove()` operator is used, I am forced to use a whole list of risky hiding rules: 
`24boxing.com.ua,4mama.ua,autoua.net,avtovod.com.ua,beauty.ua,besplatka.ua,bigmir.net,bilshe.com,buhgalter.com.ua,buhgalter911.com,businessua.com,censor.net,dengi.ua,ditey.com,enovosty.com,epravda.com.ua,eurointegration.com.ua,f1analytic.com,facenews.ua,factor.ua,football-ukraine.com,football24.ua,footballtransfer.com.ua,forumodua.com,gazeta.ua,gismeteo.ua,glianec.com,gorod.dp.ua,hvylya.net,i.ua,inforesist.org,internetua.com,isport.ua,kolobok.ua,kp.ua,kriminal.tv,lifedon.com.ua,liga.net,mama.ua,meteo.ua,mport.ua,nashamama.com,newsyou.info,nnovosti.info,nv.ua,okino.ua,orakul.com,ostro.org,panno4ka.net,pravda.com.ua,real-vin.com,sinoptik.ua,smak.ua,sportanalytic.com,stravy.net,tochka.net,toneto.net,tv.ua,tvoymalysh.com.ua,u-news.com.ua,udoktora.net,vsetv.com,zdorovia.com.ua##style:first-child + div[class]:not([class*=" "]):not([class*="-"]):not([class*="_"]):not(.container)
24boxing.com.ua,4mama.ua,autoua.net,avtovod.com.ua,beauty.ua,besplatka.ua,bigmir.net,bilshe.com,buhgalter.com.ua,buhgalter911.com,businessua.com,censor.net,dengi.ua,ditey.com,enovosty.com,epravda.com.ua,eurointegration.com.ua,f1analytic.com,facenews.ua,factor.ua,football-ukraine.com,football24.ua,footballtransfer.com.ua,forumodua.com,gazeta.ua,gismeteo.ua,glianec.com,gorod.dp.ua,hvylya.net,i.ua,inforesist.org,internetua.com,isport.ua,kolobok.ua,kp.ua,kriminal.tv,lifedon.com.ua,liga.net,mama.ua,meteo.ua,mport.ua,nashamama.com,newsyou.info,nnovosti.info,nv.ua,okino.ua,orakul.com,ostro.org,panno4ka.net,pravda.com.ua,real-vin.com,sinoptik.ua,smak.ua,sportanalytic.com,stravy.net,tochka.net,toneto.net,tv.ua,tvoymalysh.com.ua,u-news.com.ua,udoktora.net,vsetv.com,zdorovia.com.ua##style:first-child ~ slot ~ div[class]:not([class*=" "]):not([class*="_"])`

It is difficult to find such rules on ukr.net, so redirect rules are used there.

It would be great if you could modify the `remove()` operator in such a way that when visiting pages opened in the background tab, there would be no unblocked ads.

This issue does not require immediate resolution, but please keep it in mind for the future if possible.

**Reply by gorhill** [reply]
My last comment says it's fixed, the procedural filters were first applied only after  a DOM mutation in uBOL -- this was a mistake in the new code.

If you have a case for uBO where there is an issue with cosmetic filters in background tab, just provide repro steps so that I will investigate what is the real cause (I was wrong above for uBOL, the issue was elsewhere than usage of requestAnimationFrame).

**Reply by dimisa-RUAdList** [reply]
For uBO: `https://docs-python.ru/tutorial/ustanovka-python/ustanovka-windows/`
Open in the background tab any link from the sidebar. Do not go to this tab right away, wait a few seconds.

**Reply by gorhill** [reply]
What am I supposed to see or not see?

**Reply by dimisa-RUAdList** [reply]
Due to the fact that the `docs-python.ru##div[id^="yandex_rtb_"]:remove()` rule does not work in the background tab, the detector is triggered and part of the interface disappears. Refresh this page and you will see how it should be.

**Comment by uBlock-user** [comment]
> and probably it does not fire when the page is not visible

[requestIdleCallback](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback) doesn't suffer from that, can be used as a replacement.

**Comment by z0ccc** [comment]
Will importing filter lists be an option for ublock lite? If not, why? Thanks

**Reply by gothic-bum** [reply]
https://github.com/uBlockOrigin/uBOL-issues/issues/6#issuecomment-1314154689

**Comment by uBlock-user** [comment]
@gorhill Will uBOL ever get dynamic filtering ? That's the only thing stopping me from using UBOL in chrome.

**Reply by Yuki2718** [reply]
Didn't he say it's impobbile as MV3 lacks a way to implement noop (falling to static filers) in reddit?

**Reply by gorhill** [reply]
Dynamic filtering requires the listing of all hostnames for which there was a network request, which means a webRequest listener, which means the service worker would be constantly be brought to life, which would make uBOL no longer declarative. Beside, uBOL would be unable to easily tell whether a network request was blocked by itself or another extension, or even just a network error.

So answer is no.

**Reply by uBlock-user** [reply]
> Didn't he say it's impobbile as MV3 lacks a way to implement noop

I don't know as I'm not on reddit anymore. I thought things must have changed in two years as many things that looked impossible have now been added to uBOL, shame looks like mv3 truly killed innovation for sure. Will be sticking to mv2 then.

**Comment by SampeiNihira** [comment]
For the time being, UBO Lite does not support cookie consent blocking.
It is possible to compensate while waiting for this functionality with an extension always with MV3 rules:

https://github.com/OhMyGuus/I-Still-Dont-Care-About-Cookies

the maintainer currently declares 1462 static rules + 1 dynamic rule per whitelisted domain.

**Comment by debsidian** [comment]
Hi @SampeiNihira : Just bringing [this thead](https://github.com/uBlockOrigin/uBOL-issues/issues/6#issuecomment-1315259285) over here as requested.

Yes, I'm aware. The whole MV2/MV3 thing caused me to begin a transition from Chromium to Firefox about a year ago. Now I'm using FF as my daily driver and only boot up chromium when there is a page rendering issue. I'd estimate I have to use chromium only once or twice a month.

For me, uBO is the "can't live without it", deal-breaker piece of software while the browser is (mostly) fungible. I get that I'm in a backwards, "tail wagging the dog" scenario, but my decision works for me, quite well.

My guess is that a few people might switch to FF over the MV2/MV3 thing, just like I did. Time will tell.

Edit to add: I made the earlier comment that I did because my personal belief is that for people who want the additional uBO functionality but unwilling to use Firefox, they have an option to make up a lot of the difference in functionality by using the extension created by Adguard which requires full permissions. So there _is_ an option for those folks. But creating a new extension with additional permissions is outside @gorhill's stated project goals for uBOL. Just like he's stated over and over again, his users don't want him to burnout over this like he did on uMatrix. So it makes sense to respect his clearly stated project boundaries.

**Comment by BaconCatBug** [comment]
Any update on cosmetic filters for UBO Lite? I installed the latest version from the chrome web store but don't see the option to enable/add any cosmetic filters.

**Reply by Yuki2718** [reply]
https://github.com/uBlockOrigin/uBOL-issues/issues/6#issuecomment-1314154689

**Reply by gorhill** [reply]
Specific cosmetic filters are supported when you enable _Optimal_ mode. Generic cosmetic filters are supported when you enable _Complete_ mode. Cosmetic filters are a subset of _extended filtering_, which also include scriptlet-based filtering.

Custom filters, either network- or cosmetic-based are not supported.

**Comment by mustafakamran98** [comment]
Not sure if this is the right place, but Google has posted an [updated timeline](https://developer.chrome.com/en/blog/resuming-the-transition-to-mv3/) for the Manifest V3 transition. I wonder if they will delay it again when deadline approaches.
